About Window

techfrapi opened the pull request:
Related to #32

karastoyanov committed:
Merge pull request #2 from Stake-And-Rope/development

gitignore file updated

karastoyanov committed:
Merge pull request #9 from Stake-And-Rope/development

Release market-app v1.0.0-alpha

techfrapi committed:
about window created

karastoyanov commented on pyqt5-gui/about.py:
```diff
@@ -0,0 +1,25 @@
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle("About")
+        self.setGeometry(650, 300, 500, 200)
```
Change the window size to 500/300pxs

karastoyanov replied:
Resolved with [10c00e7](https://github.com/Stake-And-Rope/market-app/pull/33/commits/10c00e7b104d37b4baf95ac5e1c0f8f312cb755f)

techfrapi committed:
change windows size

techfrapi committed:
added Story, Contact and Contributors layout

techfrapi commented:
Last commit https://github.com/Stake-And-Rope/market-app/commit/9f872eeb627d7feffe8952c59434c9011e6780aa
Added:
- Story About our company
- Contact us information
- Contributors - with names and link to Github

karastoyanov commented on pyqt5-gui/about.py:
```diff
@@ -15,8 +15,67 @@
-        self.setGeometry(650, 300, 500, 300)
+        self.setWindowTitle("About Us")
+        self.setGeometry(650, 300, 600, 400)
+
```
hgf

karastoyanov commented on pyqt5-gui/about.py:
```diff
@@ -0,0 +1,84 @@
+
+        main_layout = QVBoxLayout()
+
+        """Init the our story layout"""
```
Change the comments lines to upper letters only

karastoyanov commented on pyqt5-gui/about.py:
```diff
@@ -0,0 +1,84 @@
+        super().__init__()
+        self.setWindowTitle("About Us")
+        self.setGeometry(650, 300, 600, 400)
+
```
use the `.setWindowIcon()` method to set the window icon. You can use the file `market.png` in img folder

karastoyanov replied:
Import custom font using the following block of code

```python
        font = QFontDatabase.addApplicationFont(r'../fonts/jetbrains-mono.regular.ttf')
        if font >= 0:
            fonts = QFontDatabase.applicationFontFamilies(font)
        else:
            print("Error loading fonts!")
```

Then, you can set the custom font to any object below, e.g.

```python
        contact_us = QLabel()
        contact_us.setText("<b><font size='4'>Contact Us</font></b><br>"
                           "123-456-7890<br>"
                           "stake_robe@example.com")
        contact_us.setFont(QFont(fonts[0], 12)) # where 12 is the font size
```

karastoyanov reviewed (changes requested):
Check the following code reviews

karastoyanov committed:
about window minor gui rework

karastoyanov committed:
about_window major GUI rework
 - added custom font
 - text layouts contents and design rework
 - comments updated
 - custom window icon
 - new vertical layour for github button, current version and release date
 - custom button leading to github repository

karastoyanov committed:
new custom font and github icon

karastoyanov commented on pyqt5-gui/about.py:
```diff
@@ -0,0 +1,84 @@
+app = QApplication(sys.argv)
+about_window = About()
+about_window.show()
+app.exec_()
```
test code review

karastoyanov reviewed (changes requested):
take action
